Build failure: missing gtest/gtest.h

A clean build can fail with missing gtest.

I tried to add it as dependency for compilation, but at least the naive approach fails (https://github.com/boutproject/BOUT-dev/pull/1629)

Log:
https://omadesala.physics.dcu.ie/BEST/2019-03-15_02:00/build.log

Relevant part:
```
  Compiling  invert/test_fft.cxx
invert/test_fft.cxx:1:10: fatal error: gtest/gtest.h: No such file or directory
 #include "gtest/gtest.h"
          ^~~~~~~~~~~~~~~
compilation terminated.
make[1]: *** [makefile:102: invert/test_fft.o] Error 1
make: *** [makefile:60: build-check-unit-tests] Error 2the last 5 days.
```

I could reproduce this using docker:
```
 podman run  -it --rm fedora:29 bash
dnf install -y git m4 zlib-devel autoconf autoconf-archive  gettext-devel automake environment-modules netcdf-devel netcdf-cxx4-devel hdf5-devel fftw-devel make python3 python3-numpy python3-Cython python3-netcdf4 python3-scipy blas-devel lapack-devel gcc-c++ python3-jinja2 doxygen python3-sphinx mpich-devel openmpi-devel
git clone https://github.com/boutproject/BOUT-dev.git
cd BOUT-dev/ && source /usr/share/Modules/init/bash && module load mpi/mpich-x86_64
git checkout origin/next && ./configure --enable-shared && make -j 2 build-check
```

I haven't tested further, but it seems to depend on the number of threads used for building, on whether it succeeds or not ...

--- a/tests/unit/makefile
+++ b/tests/unit/makefile
@@ -43,7 +43,7 @@ clean ::
 
 # Download Google Test
 $(GTEST_SENTINEL) :
-	echo "Downloading Google Test"
+	@echo "Downloading Google Test"
 	git submodule update --init --recursive
 
 # For simplicity and to avoid depending on Google Test's
@@ -78,6 +78,8 @@ $(TARGET): makefile $(BOUT_TOP)/make.config $(OBJ) $(SUB_LIBS) bout_test_main.a
 
 TEST_SOURCES = $(shell find $(BOUT_TEST_DIR) -type f -name "test_*.cxx" 2> /dev/null)
 TEST_OBJECTS = $(TEST_SOURCES:%.cxx=%.o)
+
+$(TEST_OBJECTS): | $(GTEST_SENTINEL)
 
 serial_tests: makefile $(BOUT_TOP)/make.config $(OBJ) $(LIB) checklib \
               $(SUB_LIBS) $(TEST_OBJECTS) bout_test_main.a

